### PR TITLE
Gromm/preview form

### PR DESCRIPTION
**Pull request by guy-ro**

### CODE REVIEW

File: src/Components/FormCreation/GetPreviewLinkDialog.js [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+          viewBox="0 0 24 24"
+          fill="none"
+          xmlns="http://www.w3.org/2000/svg"
+        >
+          <path
+            fill-rule="evenodd"
+            clip-rule="evenodd"
+            d="M12 2C8.14 2 5 5.14 5 9C5 11.38 6.19 13.47 8 14.74V17C8 17.55 8.45 18 9 18H15C15.55 18 16 17.55 16 17V14.74C17.81 13.47 19 11.38 19 9C19 5.14 15.86 2 12 2ZM9 21C9 21.55 9.45 22 10 22H14C14.55 22 15 21.55 15 21V20H9V21ZM14 13.7L14.85 13.1C16.2 12.16 17 10.63 17 9C17 6.24 14.76 4 12 4C9.24 4 7 6.24 7 9C7 10.63 7.8 12.16 9.15 13.1L10 13.7V16H14V13.7Z"
+            fill="black"
+            fill-opacity="0.54"
```
Comment (ahmedhamodi) [inline src/Components/FormCreation/GetPreviewLinkDialog.js:189]: Should be fillRule, clipRule, and fillOpacity. See travis CI failure.

**Review by ahmedhamodi — COMMENTED** [review]
Besides the comments mentioned at the demo today, there are some travis CI failures still! I have outlined the lines, but here is the screenshot too:

image

File: src/Components/FormCreation/GetPreviewLinkDialog.js (hunk 1 above)
Comment (guy-ro) [reply src/Components/FormCreation/GetPreviewLinkDialog.js]: Fixed. Also all comments during demo were already fixed.

File: src/Components/FormCreation/CreateEditFormHeader.tsx [hunk 2]
```diff
@@ -35,44 +38,162 @@ const DescriptionInput = styled(InputBase)`
...
   onChange: (data: HeaderData) => void;
 };
 
 function CreateEditFormHeader({
   name,
   description,
+  previewLink,
   onChange
 }: Props): React.ReactElement<typeof Header> {
   const onNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (GodGreg) [inline src/Components/FormCreation/CreateEditFormHeader.tsx:86]: I changed the logic slightly for this; I will handle the change when i rebase my pr

File: src/Components/FormCreation/GetPreviewLinkDialog.js [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+  position: absolute;
+  left: 11.65%;
+  right: 25.12%;
+  top: 106px;
+  font-family: Roboto;
+  font-size: 14px;
+  font-style: normal;
+  font-weight: 400;
+  line-height: 21px;
+  letter-spacing: 0.25px;
```
Comment (GodGreg) [inline src/Components/FormCreation/GetPreviewLinkDialog.js:46]: I also created some global CSS definitions in my PR

It has 
`
  font-family: Roboto;
  font-size: 14px;
  font-style: normal;
  font-weight: 400;
  line-height: 21px;
  letter-spacing: 0.25px;
`

File: src/Components/FormCreation/GetPreviewLinkDialog.js [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+          applicants to apply. Response submitted with this link <b>will not</b>{" "}
+          be saved.
+        </WarningMessageText>
+      </WarningMessage>
+      <PreviewLinkWrapper>
+        <p>Preview Link</p>
+        <input value={link} onClick={handleFocus} />
+        <div
+          style={{
+            left: "0%",
```
Comment (GodGreg) [inline src/Components/FormCreation/GetPreviewLinkDialog.js:203]: Styling should be defined at the top and not inside `return`

File: src/Components/FormCreation/GetPreviewLinkDialog.js [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+        >
+          Open In New Window
+        </Button>
+        <Button
+          onClick={copyLinkToClipboard}
+          href="#text-buttons"
+          variant="outlined"
+          color="primary"
+          style={{
+            marginLeft: "23px",
```
Comment (GodGreg) [inline src/Components/FormCreation/GetPreviewLinkDialog.js:238]: styling should not be defined here

File: src/Components/FormCreation/GetPreviewLinkDialog.js [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+      </PreviewLinkWrapper>
+      <ButtonWrapper>
+        <Button
+          onClick={() => {
+            window.open(link);
+          }}
+          href="#text-buttons"
+          color="primary"
+          style={{
+            fontFamily: "Roboto",
```
Comment (GodGreg) [inline src/Components/FormCreation/GetPreviewLinkDialog.js:220]: styling should not be defined here

**Review by GodGreg — CHANGES_REQUESTED** [review]
Overall well done,

I added some notes of still I will change when i rebase my PR. 

The request changes is changing where the CSS is defined. We haven't been defining it inside `style={{}}` for most of the app unless it is a 1 off thing. In your button wrapper you can override material-ui-button css. (the classname can be found with inspect element)

File: src/Components/FormCreation/GetPreviewLinkDialog.js (hunk 3 above)
Comment (GodGreg) [reply src/Components/FormCreation/GetPreviewLinkDialog.js:46]: I will also handle this

**Review by GodGreg — APPROVED** [review]

**Pull request merged**